jf_agent/git/__init__.py

```diff
@@ -253,16 +253,17 @@ def get_git_client(config: GitConfig, git_creds: dict, skip_ssl_verification: bo
             )
 
     except Exception as e:
-        agent_logging.log_and_print(
+        agent_logging.log_and_print_error_or_warning(
             logger,
             logging.ERROR,
-            f'Failed to connect to {config.git_provider}:\n{e}',
+            msg_args=[config.git_provider, e],
+            error_code=2101,
             exc_info=True,
         )
         return
 
     # if the git provider is none of the above, throw an error
-    raise ValueError(f'unsupported git provider {config.git_provider}')
+    raise ValueError(f'unsupported git provider {config.git_provider}') # Customer Config Error
```

Comment:
I think you can yank the `# Customer Config Error` annotation you added here, since we're going to be doing that categorization with the alternate approach